### PR TITLE
Improve responsiveness of synchronization

### PR DESCRIPTION
**Pull request by n-wach**
We want to make the synchronization feel more smooth:

- We increased the frequency of sync to 10 times per second (this can be reduced).

- In order to be able to update the time when a user seeks the video, the timer needs to be always running. We added a check to ensure we don't unnecessarily seek (if the time has changed since last seek).

Although the timer is always running, CPU usage remains at 0% when the video is paused.

### CODE REVIEW

File: src/main/webapp/synchronizer.js [hunk 1]
```diff
@@ -25,16 +27,15 @@ function startVideoSyncTimer() {
...
 
-/** Stops video sync timer. */
-function stopVideoSyncTimer() {
-  clearInterval(videoSyncTimer);
-}
-
 /**
  * Calls functions that seek transcript, and discussion to {@code currentTime}
  * (number of seconds since video started playing).
  */
```
Comment (amuamushu) [inline src/main/webapp/synchronizer.js:33]: can you update the JSDoc to reflect how the calls are not made if the time has not changed?

Comment (n-wach) [reply src/main/webapp/synchronizer.js:33]: Done, thanks!

File: src/main/webapp/synchronizer.js [hunk 2]
```diff
@@ -13,7 +13,9 @@
 // limitations under the License.
 
 let videoSyncTimer;
-const TIME_INTERVAL = 1000;
+let lastTime;
+
+const TIME_INTERVAL = 100;
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:18]: can you rename it to TIME_INTERVAL_MS so we know this is milliseconds.

**Comment by n-wach** [comment]
> Why aren't we stopping the timer when the video is paused any more?
@f-camp 

Unfortunately there isn't an event for when the user drags the little slider around, and we think it is important to sync in this case.  The only way is to constantly poll `.getCurrentTime()`.  But with the `last == current` check we don't do anything unless the time actually changed

File: src/main/webapp/synchronizer.js (hunk 2 above)
Comment (n-wach) [reply src/main/webapp/synchronizer.js:18]: Done!

**Pull request merged**